### PR TITLE
SCLabelPolicy

### PR DESCRIPTION
**Pull request by desmay**
SCLabelPolicy
#### Pull Request (PR) description
New resource SCLabelPolicy
Updates to SPOSiteDesign to support AzureAD app 
#### This Pull Request (PR) fixes the following issues
Fixes 758

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/microsoft365dsc/791)

### CODE REVIEW

**Review by NikCharlebois — CHANGES_REQUESTED** [review]
Few items to address

> Reviewed 11 of 11 files at r1.
> *[Reviewable](https://reviewable.io/reviews/microsoft/microsoft365dsc/791#-:-MHpd0_l3KIsuBxX_yfx:bs3iokc)* status: all files reviewed, 7 unresolved discussions (waiting on @desmay)

___
*[Modules/Microsoft365DSC/DSCResources/MSFT_SCLabelPolicy/MSFT_SCLabelPolicy.psm1, line 264 at r1](https://reviewable.io/reviews/microsoft/microsoft365dsc/791#-MHpazIvEUH57VxOXLAq:-MHpazIw06zpgKZDmBuD:bb1v6a7) ([raw file](https://github.com/microsoft/microsoft365dsc/blob/13ff9d2278452b3288b60da050027cd9cddf5e83/Modules/Microsoft365DSC/DSCResources/MSFT_SCLabelPolicy/MSFT_SCLabelPolicy.psm1#L264)):*
> ```PowerShell
>             $CreationParams["AdvancedSettings"] = $advanced
>         }
>         #Remove parameters not used in Set-LabelPolicy
> ```

Do we mean New-LabelPolicy here?
___
*[Modules/Microsoft365DSC/DSCResources/MSFT_SCLabelPolicy/MSFT_SCLabelPolicy.psm1, line 281 at r1](https://reviewable.io/reviews/microsoft/microsoft365dsc/791#-MHpb16Z6h2fP7l3NHJy:-MHpb16Z6h2fP7l3NHJz:b3pgk4x) ([raw file](https://github.com/microsoft/microsoft365dsc/blob/13ff9d2278452b3288b60da050027cd9cddf5e83/Modules/Microsoft365DSC/DSCResources/MSFT_SCLabelPolicy/MSFT_SCLabelPolicy.psm1#L281)):*
> ```PowerShell
>         try
>         {
>             New-LabelPolicy @CreationParams
> ```

Since we can pass in all the other params that were removed (e.g.: AddLabels, etc.)  should we call into the Set-LabelPolicy right after the New? So the scenarios would be:

New --> Call New-LabelPolicy then Set-LabelPolicy
Update --> Call Set-LabelPolicy only
Remove --> Call Remove-LabelPolicy
___
*[Modules/Microsoft365DSC/DSCResources/MSFT_SCLabelPolicy/MSFT_SCLabelPolicy.schema.mof, line 12 at r1](https://reviewable.io/reviews/microsoft/microsoft365dsc/791#-MHpc6OMBjcVuNdkQ9SU:-MHpc6OMBjcVuNdkQ9SV:b-3usncp) ([raw file](https://github.com/microsoft/microsoft365dsc/blob/13ff9d2278452b3288b60da050027cd9cddf5e83/Modules/Microsoft365DSC/DSCResources/MSFT_SCLabelPolicy/MSFT_SCLabelPolicy.schema.mof#L12)):*
> ```
> {
>     [Key, Description("The Name parameter specifies the unique name for the sensitivity label. The maximum length is 64 characters. If the value contains spaces, enclose the value in quotation marks.")] String Name;
>     [Write, Description("Specify if this rule should exist or not."), ValueMap{"Present","Absent"}, Values{"Present","Absent"}] String Ensure;
> ```

Should be Policy
___
*[Modules/Microsoft365DSC/DSCResources/MSFT_SPOSiteScript/MSFT_SPOSiteScript.psm1, line 88 at r1](https://reviewable.io/reviews/microsoft/microsoft365dsc/791#-MHpcMoB8AGIzYR-qUKW:-MHpcMoB8AGIzYR-qUKX:b-m0l2eg) ([raw file](https://github.com/microsoft/microsoft365dsc/blob/13ff9d2278452b3288b60da050027cd9cddf5e83/Modules/Microsoft365DSC/DSCResources/MSFT_SPOSiteScript/MSFT_SPOSiteScript.psm1#L88)):*
> ```PowerShell
>             $SiteScript = $null
>             ##### Check to see if more than one site script is returned
> <<<<<<< HEAD
> ```

Looks like you have merge trace still in. Remove the 3 lines
___
*[Modules/Microsoft365DSC/DSCResources/MSFT_SPOSiteScript/MSFT_SPOSiteScript.psm1, line 177 at r1](https://reviewable.io/reviews/microsoft/microsoft365dsc/791#-MHpcYx80I47xOx6JvsA:-MHpcYx80I47xOx6JvsB:b-n0whos) ([raw file](https://github.com/microsoft/microsoft365dsc/blob/13ff9d2278452b3288b60da050027cd9cddf5e83/Modules/Microsoft365DSC/DSCResources/MSFT_SPOSiteScript/MSFT_SPOSiteScript.psm1#L177)):*
> ```PowerShell
>         $CertificateThumbprint,
> 
>         [Parameter(Mandatory = $true)]
> ```

It shouldn't be mandatory across the board anymore since we can also use other auth mechanism
___
*[Modules/Microsoft365DSC/DSCResources/MSFT_SPOSiteScript/MSFT_SPOSiteScript.psm1, line 341 at r1](https://reviewable.io/reviews/microsoft/microsoft365dsc/791#-MHpcqQs4mHmTi1b2B4l:-MHpcqQs4mHmTi1b2B4m:b-mw1sh7) ([raw file](https://github.com/microsoft/microsoft365dsc/blob/13ff9d2278452b3288b60da050027cd9cddf5e83/Modules/Microsoft365DSC/DSCResources/MSFT_SPOSiteScript/MSFT_SPOSiteScript.psm1#L341)):*
> ```PowerShell
>         $CertificateThumbprint,
> 
>         [Parameter(Mandatory = $true)]
> ```

Should be optional
___
*[Modules/Microsoft365DSC/DSCResources/MSFT_SPOSiteScript/MSFT_SPOSiteScript.psm1, line 370 at r1](https://reviewable.io/reviews/microsoft/microsoft365dsc/791#-MHpcsnS8XNMEix5_ZDc:-MHpcsnS8XNMEix5_ZDd:b-mw1sh7) ([raw file](https://github.com/microsoft/microsoft365dsc/blob/13ff9d2278452b3288b60da050027cd9cddf5e83/Modules/Microsoft365DSC/DSCResources/MSFT_SPOSiteScript/MSFT_SPOSiteScript.psm1#L370)):*
> ```PowerShell
>     param
>     (
>         [Parameter(Mandatory = $true)]
> ```

Should be optional

**Review by desmay — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/microsoft/microsoft365dsc/791)* status: all files reviewed, 7 unresolved discussions (waiting on @NikCharlebois)

___
*[Modules/Microsoft365DSC/DSCResources/MSFT_SCLabelPolicy/MSFT_SCLabelPolicy.psm1, line 264 at r1](https://reviewable.io/reviews/microsoft/microsoft365dsc/791#-MHpazIvEUH57VxOXLAq:-MHpwOYy4dllmLljHn-v:b-896fix) ([raw file](https://github.com/microsoft/microsoft365dsc/blob/13ff9d2278452b3288b60da050027cd9cddf5e83/Modules/Microsoft365DSC/DSCResources/MSFT_SCLabelPolicy/MSFT_SCLabelPolicy.psm1#L264)):*
<details><summary><i>Previously, NikCharlebois (Nik Charlebois) wrote…</i></summary><blockquote>

Do we mean New-LabelPolicy here?
</blockquote></details>

Done.
___
*[Modules/Microsoft365DSC/DSCResources/MSFT_SCLabelPolicy/MSFT_SCLabelPolicy.psm1, line 281 at r1](https://reviewable.io/reviews/microsoft/microsoft365dsc/791#-MHpb16Z6h2fP7l3NHJy:-MHpwRy24u86FonRBQDp:b-896fix) ([raw file](https://github.com/microsoft/microsoft365dsc/blob/13ff9d2278452b3288b60da050027cd9cddf5e83/Modules/Microsoft365DSC/DSCResources/MSFT_SCLabelPolicy/MSFT_SCLabelPolicy.psm1#L281)):*
<details><summary><i>Previously, NikCharlebois (Nik Charlebois) wrote…</i></summary><blockquote>

Since we can pass in all the other params that were removed (e.g.: AddLabels, etc.)  should we call into the Set-LabelPolicy right after the New? So the scenarios would be:

New --> Call New-LabelPolicy then Set-LabelPolicy
Update --> Call Set-LabelPolicy only
Remove --> Call Remove-LabelPolicy
</blockquote></details>

Done.
___
*[Modules/Microsoft365DSC/DSCResources/MSFT_SCLabelPolicy/MSFT_SCLabelPolicy.schema.mof, line 12 at r1](https://reviewable.io/reviews/microsoft/microsoft365dsc/791#-MHpc6OMBjcVuNdkQ9SU:-MHpjDFR4ls1ayI5ktvJ:b-896fix) ([raw file](https://github.com/microsoft/microsoft365dsc/blob/13ff9d2278452b3288b60da050027cd9cddf5e83/Modules/Microsoft365DSC/DSCResources/MSFT_SCLabelPolicy/MSFT_SCLabelPolicy.schema.mof#L12)):*
<details><summary><i>Previously, NikCharlebois (Nik Charlebois) wrote…</i></summary><blockquote>

Should be Policy
</blockquote></details>

Done.
___
*[Modules/Microsoft365DSC/DSCResources/MSFT_SPOSiteScript/MSFT_SPOSiteScript.psm1, line 88 at r1](https://reviewable.io/reviews/microsoft/microsoft365dsc/791#-MHpcMoB8AGIzYR-qUKW:-MHpk3AMBS2wRMvbtWaG:b-896fix) ([raw file](https://github.com/microsoft/microsoft365dsc/blob/13ff9d2278452b3288b60da050027cd9cddf5e83/Modules/Microsoft365DSC/DSCResources/MSFT_SPOSiteScript/MSFT_SPOSiteScript.psm1#L88)):*
<details><summary><i>Previously, NikCharlebois (Nik Charlebois) wrote…</i></summary><blockquote>

Looks like you have merge trace still in. Remove the 3 lines
</blockquote></details>

Done.
___
*[Modules/Microsoft365DSC/DSCResources/MSFT_SPOSiteScript/MSFT_SPOSiteScript.psm1, line 177 at r1](https://reviewable.io/reviews/microsoft/microsoft365dsc/791#-MHpcYx80I47xOx6JvsA:-MHpkO7o5W0cmdElI5ce:b-896fix) ([raw file](https://github.com/microsoft/microsoft365dsc/blob/13ff9d2278452b3288b60da050027cd9cddf5e83/Modules/Microsoft365DSC/DSCResources/MSFT_SPOSiteScript/MSFT_SPOSiteScript.psm1#L177)):*
<details><summary><i>Previously, NikCharlebois (Nik Charlebois) wrote…</i></summary><blockquote>

It shouldn't be mandatory across the board anymore since we can also use other auth mechanism
</blockquote></details>

Done.
___
*[Modules/Microsoft365DSC/DSCResources/MSFT_SPOSiteScript/MSFT_SPOSiteScript.psm1, line 341 at r1](https://reviewable.io/reviews/microsoft/microsoft365dsc/791#-MHpcqQs4mHmTi1b2B4l:-MHpkQ8z9Zg7vMvuLwpG:b-896fix) ([raw file](https://github.com/microsoft/microsoft365dsc/blob/13ff9d2278452b3288b60da050027cd9cddf5e83/Modules/Microsoft365DSC/DSCResources/MSFT_SPOSiteScript/MSFT_SPOSiteScript.psm1#L341)):*
<details><summary><i>Previously, NikCharlebois (Nik Charlebois) wrote…</i></summary><blockquote>

Should be optional
</blockquote></details>

Done.
___
*[Modules/Microsoft365DSC/DSCResources/MSFT_SPOSiteScript/MSFT_SPOSiteScript.psm1, line 370 at r1](https://reviewable.io/reviews/microsoft/microsoft365dsc/791#-MHpcsnS8XNMEix5_ZDc:-MHpkRQyBHG3N2x8GQgG:b-896fix) ([raw file](https://github.com/microsoft/microsoft365dsc/blob/13ff9d2278452b3288b60da050027cd9cddf5e83/Modules/Microsoft365DSC/DSCResources/MSFT_SPOSiteScript/MSFT_SPOSiteScript.psm1#L370)):*
<details><summary><i>Previously, NikCharlebois (Nik Charlebois) wrote…</i></summary><blockquote>

Should be optional
</blockquote></details>

Done.

**Review by NikCharlebois — APPROVED** [review]
:lgtm:

> Reviewed 4 of 4 files at r2.
> *[Reviewable](https://reviewable.io/reviews/microsoft/microsoft365dsc/791#-:-MHq-EpqAVXLQJgR9pPl:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved
